### PR TITLE
feat: Add Autopilot test

### PR DESCRIPTION
**Pull request by kuisathaverat**
## What does this PR do?

It adds a new workflow to execute a test in GKE Autopilot clusters. The test uses the last 2 versions of GKE in the stable channel.

## Why is it important?

The test verifies the proper behaviour of the Elastic Agent+Kubernetes integration on GKE Autopilot clusters.

## Checklist

- [X] My code follows the style guidelines of this project
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- ~[ ] I have made corresponding change to the default configuration files~
- ~[ ] I have added tests that prove my fix is effective or that my feature works~
- ~[ ] I have added an entry in `./changelog/fragments` using the [changelog tool](https://github.com/elastic/elastic-agent#changelog)~
- ~[ ] I have added an integration test or an E2E test~

### CODE REVIEW

File: .github/workflows/autopilot-test.yml [hunk 1]
```diff
@@ -39,35 +55,55 @@ jobs:
...
         run: |
           ./docs/scripts/autopilot-tests.sh "${CLUSTER_NAME}-default" "autopilot-integration"
+
       - name: Destroy cluster
         uses: elastic/apm-pipeline-library/.github/actions/oblt-cli@current
         if: always()
         with:
-          command: |
-            oblt-cli cluster destroy --cluster-name "${{ env.CLUSTER_NAME }}" --force
+          command: cluster destroy --cluster-name "${{ env.CLUSTER_NAME }}" --force
```
Comment (v1v) [inline .github/workflows/autopilot-test.yml:108]: NIT:
Do you think this should not cause any errors if for any reason it failed?

**Review by v1v — APPROVED** [review]

File: .github/workflows/autopilot-test.yml [hunk 2]
```diff
@@ -0,0 +1,109 @@
+---
+name: Autopilot test
+
+on:
+  workflow_dispatch:
+
```
Comment (kuisathaverat) [inline .github/workflows/autopilot-test.yml:6]: TODO add schedule

File: .github/workflows/autopilot-test.yml [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+          CLUSTER_NAME=$(jq -r .ClusterName $PWD/cluster_info.json)
+          echo "CLUSTER_NAME=${CLUSTER_NAME}" >> $GITHUB_ENV
+
+      - name: Get cluster credentials
+        run: |
+          gcloud container clusters get-credentials "${CLUSTER_NAME}" --zone "${ZONE}" --project "${PROJECT_ID}"
+
+      - name: Run tests
+        run: |
+          ./docs/scripts/autopilot-tests.sh "${CLUSTER_NAME}-default" "autopilot-integration"
```
Comment (kuisathaverat) [inline .github/workflows/autopilot-test.yml:102]: it depends on https://github.com/elastic/elastic-agent/pull/1961

**Comment by pierrehilbert** [comment]
@kuisathaverat can we merge this one?

**Comment by kuisathaverat** [comment]
@pierrehilbert I guess so

**Comment by jlind23** [comment]
Closing this for now until it becomes a priority again.

**Pull request closed without merging**